skorch/callbacks/lr_scheduler.py

```diff
@@ -158,19 +158,20 @@ def on_epoch_end(self, net, **kwargs):
             if callable(self.monitor):
                 score = self.monitor(net)
             else:
-                if epoch:
-                    score = net.history[-2, self.monitor]
-                else:
-                    if self.lr_scheduler_.mode == 'max':
-                        score = -np.inf
-                    else:
-                        score = np.inf
+                if self.lr_scheduler_.mode == 'max':
+                    score = -np.inf
+                elif self.lr_scheduler_.mode == 'min':
+                    score = np.inf
+                elif epoch:
+                    score = net.history[-1, self.monitor]
```

Comment:
I'm no expert on lr schedulers, so can't comment on the exact logic that should be applied here. However, what I'm seeing is that the logic might have a "hole", where score wouldn't be defined at all if none of the `if` or `elif` matches. In the logic before, score would always be defined.

Could it be that we don't actually need `elif epoch`? Previously, we accessed `net.history[-2, self.monitor]`, i.e. the second to last row in the `history`. Therefore, we had to make sure that we're not in the first epoch (at least that's my interpretation of things). Now that we're accessing `net.history[-1, self.monitor]`, we can probably never get an index error here, hence don't need the guard. Would that make sense?

Reply:
Makes sense. Pushed a change.